### ISSUE TITLE
'packages/flutter/src/gestures/multidrag.dart': Failed assertion: line 181: 'state != null' is not true.

### ISSUE THREAD

**Issue by apwilson** [flutter/flutter#1927]
```dart
android: I/flutter : -- EXCEPTION CAUGHT BY GESTURE LIBRARY ---------------------------------
android: I/flutter : The following exception was raised while dispatching a pointer event:
android: I/flutter : 'packages/flutter/src/gestures/multidrag.dart': Failed assertion: line 181: 'state != null' is not true.
android: I/flutter : Event:
android: I/flutter : PointerUpEvent(Point(1199.5, 158.0))
android: I/flutter : Target:
android: I/flutter : Instance of 'WidgetFlutterBinding'
android: I/flutter : Stack trace:
android: I/flutter : #0      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:27)
android: I/flutter : #1      MultiDragGestureRecognizer.acceptGesture (packages/flutter/src/gestures/multidrag.dart:181)
android: I/flutter : #2      GestureArena.sweep (packages/flutter/src/gestures/arena.dart:105)
android: I/flutter : #3      BindingBase&Scheduler&Gesturer.handleEvent (packages/flutter/src/gestures/binding.dart:121)
android: I/flutter : #4      BindingBase&Scheduler&Gesturer.dispatchEvent (packages/flutter/src/gestures/binding.dart:96)
android: I/flutter : #5      BindingBase&Scheduler&Gesturer._handlePointerEvent (packages/flutter/src/gestures/binding.dart:71)
android: I/flutter : #6      BindingBase&Scheduler&Gesturer._handlePointerPacket (packages/flutter/src/gestures/binding.dart:43)
android: I/flutter : #7      _dispatchPointerPacket (file:///home/travis/build/flutter/buildbot/src/out/android_Release/gen/sky/bindings/hooks.dart:43)
android: I/flutter : ------------------------------------------------------------------------
```

**Comment by apwilson** [comment]
Another stack trace:

```dart
android: I/flutter : -- EXCEPTION CAUGHT BY GESTURE LIBRARY ---------------------------------
android: I/flutter : The following exception was raised while routing a pointer event:
android: I/flutter : 'packages/flutter/src/gestures/multidrag.dart': Failed assertion: line 181: 'state != null' is not true.
android: I/flutter : Event:
android: I/flutter : PointerUpEvent(Point(1128.0, 211.0))
android: I/flutter : Stack trace:
android: I/flutter : #0      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:27)
android: I/flutter : #1      MultiDragGestureRecognizer.acceptGesture (packages/flutter/src/gestures/multidrag.dart:181)
android: I/flutter : #2      GestureArena._tryToResolveArena (packages/flutter/src/gestures/arena.dart:138)
android: I/flutter : #3      GestureArena._resolve (packages/flutter/src/gestures/arena.dart:155)
android: I/flutter : #4      GestureArenaEntry.resolve (packages/flutter/src/gestures/arena.dart:45)
android: I/flutter : #5      OneSequenceGestureRecognizer.resolve (packages/flutter/src/gestures/recognizer.dart:80)
android: I/flutter : #6      _DragGestureRecognizer.didStopTrackingLastPointer (packages/flutter/src/gestures/drag.dart:103)
android: I/flutter : #7      OneSequenceGestureRecognizer.stopTrackingPointer (packages/flutter/src/gestures/recognizer.dart:103)
android: I/flutter : #8      OneSequenceGestureRecognizer.stopTrackingIfPointerNoLongerDown (packages/flutter/src/gestures/recognizer.dart:108)
android: I/flutter : #9      _DragGestureRecognizer.handleEvent (packages/flutter/src/gestures/drag.dart:86)
android: I/flutter : #10     PointerRouter.route (packages/flutter/src/gestures/pointer_router.dart:65)
android: I/flutter : #11     BindingBase&Scheduler&Gesturer.handleEvent (packages/flutter/src/gestures/binding.dart:117)
android: I/flutter : #12     BindingBase&Scheduler&Gesturer.dispatchEvent (packages/flutter/src/gestures/binding.dart:96)
android: I/flutter : #13     BindingBase&Scheduler&Gesturer._handlePointerEvent (packages/flutter/src/gestures/binding.dart:71)
android: I/flutter : #14     BindingBase&Scheduler&Gesturer._handlePointerPacket (packages/flutter/src/gestures/binding.dart:43)
android: I/flutter : #15     _dispatchPointerPacket (file:///home/travis/build/flutter/buildbot/src/out/android_Release/gen/sky/bindings/hooks.dart:43)
android: I/flutter : ------------------------------------------------------------------------

```

**Comment by Hixie** [comment]
It would greatly help to have a minimised test case for this.

**Comment by apwilson** [comment]
It's kind of hard to reproduce.  I need to tap a bunch on some widgets that are draggable and have a tap gesture detector.  I'll see what I can do.

**Comment by apwilson** [comment]
Added some traces.  Culprit seems to be an acceptGesture after we've removedState for the pointer:

```dart
android: I/flutter : addPointer 126
android: I/flutter : handleEvent 126
android: I/flutter : handleEvent 126
android: I/flutter : handleEvent 126
android: I/flutter : handleEvent 126
android: I/flutter : handleEvent 126
android: I/flutter : handleEvent 126
android: I/flutter : handleEvent 126
android: I/flutter : handleEvent 126
android: I/flutter : _removeState 126
android: I/flutter : acceptGesture 126
```

**Comment by apwilson** [comment]
Better traces:
```dart
android: I/flutter : addPointer 113
android: I/flutter : handleEvent 113
android: I/flutter : handleEvent 113
android: I/flutter : handleEvent 113
android: I/flutter : handleEvent 113
android: I/flutter : handleEvent up 113
android: I/flutter : _removeState 113
android: I/flutter : acceptGesture 113

```

**Comment by apwilson** [comment]
Sounds like a timing issue to me.  Some pointer events cause the pointer to be added to a gesture arena and the callback from the gesture arena just happens to happen after a pointer up for that pointer.

**Comment by apwilson** [comment]
Another slightly different stack trace:
```dart
android: I/flutter : -- EXCEPTION CAUGHT BY GESTURE LIBRARY ---------------------------------
android: I/flutter : The following exception was raised while dispatching a pointer event:
android: I/flutter : 'packages/flutter/src/gestures/multidrag.dart': Failed assertion: line 181: 'state != null' is not true.
android: I/flutter : Event:
android: I/flutter : PointerUpEvent(Point(1182.0, 54.0))
android: I/flutter : Target:
android: I/flutter : Instance of 'WidgetFlutterBinding'
android: I/flutter : Stack trace:
android: I/flutter : #0      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:27)
android: I/flutter : #1      MultiDragGestureRecognizer.acceptGesture (packages/flutter/src/gestures/multidrag.dart:181)
android: I/flutter : #2      GestureArena.sweep (packages/flutter/src/gestures/arena.dart:105)
android: I/flutter : #3      BindingBase&Scheduler&Gesturer.handleEvent (packages/flutter/src/gestures/binding.dart:121)
android: I/flutter : #4      BindingBase&Scheduler&Gesturer.dispatchEvent (packages/flutter/src/gestures/binding.dart:96)
android: I/flutter : #5      BindingBase&Scheduler&Gesturer._handlePointerEvent (packages/flutter/src/gestures/binding.dart:71)
android: I/flutter : #6      BindingBase&Scheduler&Gesturer._handlePointerPacket (packages/flutter/src/gestures/binding.dart:43)
android: I/flutter : #7      _dispatchPointerPacket (file:///home/travis/build/flutter/buildbot/src/out/android_Release/gen/sky/bindings/hooks.dart:43)
android: I/flutter : ------------------------------------------------------------------------
```

**Comment by apwilson** [comment]
Found out an easy reproduce.  two fingers down.  One drags right, the other lifts up - assert.

**Comment by apwilson** [comment]
You're in lock - simple reproduce:

```dart
void main() {
  runApp(new Overlay(initialEntries: [
        new OverlayEntry(
            builder: (_) => new GestureDetector(
            behavior: HitTestBehavior.opaque,
            onTap: () {},
            child: new Draggable(
                child: new Container(
                    width:100.0,
                    height:100.0,
                    decoration:
                        new BoxDecoration(backgroundColor: new Color(0xFFFFFF00))),
                childWhenDragging: new Container(
                  width:100.0,
                  height:100.0,
                    decoration:
                        new BoxDecoration(backgroundColor: new Color(0xFFFF00FF))),
                feedback: new Container(
                  width:100.0,
                  height:100.0,
                    decoration:
                        new BoxDecoration(backgroundColor: new Color(0xFFFF0000)))))
                )
      ]));
}
```

**Comment by abarth** [comment]
Looks like this was fixed by 51b1550df79b1e9f97876bba6410d46d4c0b9396 .  I'll write a test for this case specifically.
